File: Tests/Test_CustomMsgBox.vb
Option Explicit On
Option Strict On
Option Compare Binary
Option Infer Off

Imports System.Threading
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace Tests
    Module Tests_CustomMsgBox
        Function Test_CustomMsgBox1() As Boolean
            Task.Run(Sub()
                         Thread.Sleep(600)
                         SendKeys.SendWait("{ENTER}")
                     End Sub)

            Dim result As DialogResult = WalkmanLib.CustomMsgBox("test")

            Return TestNumber("CustomMsgBox1", result, DialogResult.OK)
        End Function

        Function Test_CustomMsgBox2() As Boolean
            Task.Run(Sub()
                         Thread.Sleep(600)
                         SendKeys.SendWait("{ENTER}")
                     End Sub)

            Dim result As DialogResult = WalkmanLib.CustomMsgBox("test", buttons:=MessageBoxButtons.YesNoCancel)

            Return TestNumber("CustomMsgBox2", result, DialogResult.Yes)
        End Function

        Function Test_CustomMsgBox3() As Boolean
            Task.Run(Sub()
                         Thread.Sleep(600)
                         SendKeys.SendWait("{ESC}")
                     End Sub)

            Dim result As DialogResult = WalkmanLib.CustomMsgBox("test", buttons:=MessageBoxButtons.YesNoCancel)

            Return TestNumber("CustomMsgBox3", result, DialogResult.Cancel)
        End Function

        Function Test_CustomMsgBox4() As Boolean
            Task.Run(Sub()
                         Thread.Sleep(600)
                         SendKeys.SendWait("{ENTER}")
                     End Sub)

            Dim result As DialogResult = WalkmanLib.CustomMsgBox("test", buttons:=MessageBoxButtons.AbortRetryIgnore)

            Return TestNumber("CustomMsgBox4", result, DialogResult.Abort)
        End Function

        Function Test_CustomMsgBox5() As Boolean
            Task.Run(Sub()
                         Thread.Sleep(600)
                         SendKeys.SendWait("{ESC}")
                     End Sub)

            Dim result As DialogResult = WalkmanLib.CustomMsgBox("test", buttons:=MessageBoxButtons.AbortRetryIgnore)

            Return TestNumber("CustomMsgBox5", result, DialogResult.Ignore)
        End Function

        Function Test_CustomMsgBox6() As Boolean
            Task.Run(Sub()
                         Thread.Sleep(600)
                         SendKeys.SendWait("{ENTER}")
                     End Sub)

            Dim result As String = WalkmanLib.CustomMsgBox("test", Nothing, "Test Button One", "Test Button Two")

            Return TestString("CustomMsgBox6", result, "Test Button One")
        End Function

        Function Test_CustomMsgBox7() As Boolean
            Task.Run(Sub()
                         Thread.Sleep(700)
                         SendKeys.SendWait("{ESC}")
                     End Sub)

            Dim result As String = WalkmanLib.CustomMsgBox("test", Nothing, "Test Button One", "Test Button Two", "Test Button Three")

            Return TestString("CustomMsgBox7", result, "Test Button Three")
        End Function

        Function Test_CustomMsgBox8() As Boolean
            Task.Run(Sub()
                         WalkmanLib.CustomMsgBox("test", "TestTitle")
                     End Sub)

            Thread.Sleep(700)
            Dim result As String = ShowPropertiesTestsHelper.GetActiveWindowText()
            SendKeys.SendWait("{ENTER}")

            Return TestString("CustomMsgBox8", result, "TestTitle")
        End Function
    End Module
End Namespace

File: Tests/Test_CustomMsgBox.cs
using System.Threading;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Tests {
    static class Tests_CustomMsgBox {
        public static bool Test_CustomMsgBox1() {
            Task.Run(() => {
                Thread.Sleep(600);
                SendKeys.SendWait("{ENTER}");
            });

            DialogResult result = WalkmanLib.CustomMsgBox("test");

            return GeneralFunctions.TestNumber("CustomMsgBox1", (int)result, (int)DialogResult.OK);
        }

        public static bool Test_CustomMsgBox2() {
            Task.Run(() => {
                Thread.Sleep(600);
                SendKeys.SendWait("{ENTER}");
            });

            DialogResult result = WalkmanLib.CustomMsgBox("test", buttons: MessageBoxButtons.YesNoCancel);

            return GeneralFunctions.TestNumber("CustomMsgBox2", (int)result, (int)DialogResult.Yes);
        }

        public static bool Test_CustomMsgBox3() {
            Task.Run(() => {
                Thread.Sleep(600);
                SendKeys.SendWait("{ESC}");
            });

            DialogResult result = WalkmanLib.CustomMsgBox("test", buttons: MessageBoxButtons.YesNoCancel);

            return GeneralFunctions.TestNumber("CustomMsgBox3", (int)result, (int)DialogResult.Cancel);
        }

        public static bool Test_CustomMsgBox4() {
            Task.Run(() => {
                Thread.Sleep(600);
                SendKeys.SendWait("{ENTER}");
            });

            DialogResult result = WalkmanLib.CustomMsgBox("test", buttons: MessageBoxButtons.AbortRetryIgnore);

            return GeneralFunctions.TestNumber("CustomMsgBox4", (int)result, (int)DialogResult.Abort);
        }

        public static bool Test_CustomMsgBox5() {
            Task.Run(() => {
                Thread.Sleep(600);
                SendKeys.SendWait("{ESC}");
            });

            DialogResult result = WalkmanLib.CustomMsgBox("test", buttons: MessageBoxButtons.AbortRetryIgnore);

            return GeneralFunctions.TestNumber("CustomMsgBox5", (int)result, (int)DialogResult.Ignore);
        }

        public static bool Test_CustomMsgBox6() {
            Task.Run(() => {
                Thread.Sleep(600);
                SendKeys.SendWait("{ENTER}");
            });

            string result = WalkmanLib.CustomMsgBox("test", null, "Test Button One", "Test Button Two");

            return GeneralFunctions.TestString("CustomMsgBox6", result, "Test Button One");
        }

        public static bool Test_CustomMsgBox7() {
            Task.Run(() => {
                Thread.Sleep(700);
                SendKeys.SendWait("{ESC}");
            });

            string result = WalkmanLib.CustomMsgBox("test", null, "Test Button One", "Test Button Two", "Test Button Three");

            return GeneralFunctions.TestString("CustomMsgBox7", result, "Test Button Three");
        }

        public static bool Test_CustomMsgBox8() {
            Task.Run(() => WalkmanLib.CustomMsgBox("test", "TestTitle"));

            Thread.Sleep(700);
            string result = ShowPropertiesTestsHelper.GetActiveWindowText();
            SendKeys.SendWait("{ENTER}");

            return GeneralFunctions.TestString("CustomMsgBox8", result, "TestTitle");
        }
    }
}
